### ISSUE TITLE
install on redhat 8.4 failed 

### ISSUE THREAD

**Issue by somewheve** [Xilinx-CNS/onload#41]
hey,  I am trying to install driver into RedHat 8.4. and the kernel is `4.18`
for some reason, i can't ask for mail support from Xilinx(because the account register failed and I ask but nobody replies me )

steps 
```
openload7.1.2.141(`by onload_install command`)   and sfutils RPM 64 bit.
```
next 
`
onload_tool reload 
` 

then
`
sfupdate --write 
`
but got no adapter. 
```
[somewheve@localhost ~]$ sfupdate
Solarflare firmware update utility [v8.2.2]
Copyright 2002-2020 Xilinx, Inc.
Loading firmware images from /usr/share/sfutils/sfupdate_images
No supported Solarflare adapters found
```
so i try find the devei  and input the  `lspci -vvv | grep "Solar"` get output  below 
```
[somewheve@localhost ~]$ lspci -vvv  | grep "Solar"
65:00.0 Ethernet controller: Solarflare Communications Device 0b13 (rev 01)
        Subsystem: Solarflare Communications Device 0000
65:00.1 Ethernet controller: Solarflare Communications Device 0b13 (rev 01)
        Subsystem: Solarflare Communications Device 0000
```

then  i try find the infomation by `modinfo sfc`  
```
filename:       /lib/modules/4.18.0-305.12.1.el8_4.x86_64/extra/sfc.ko
version:        4.15.12.1008
license:        GPL
description:    Solarflare network driver
author:         Solarflare Communications and Michael Brown <mbrown@fensystems.co.uk>
rhelversion:    8.4
srcversion:     0D781ADFE2D6C82855F0815
alias:          pci:v00001924d00001B03sv*sd*bc*sc*i*
alias:          pci:v00001924d00000B03sv*sd*bc*sc*i*
alias:          pci:v00001924d00001A03sv*sd*bc*sc*i*
alias:          pci:v00001924d00000A03sv*sd*bc*sc*i*
alias:          pci:v00001924d00001923sv*sd*bc*sc*i*
alias:          pci:v00001924d00000923sv*sd*bc*sc*i*
alias:          pci:v00001924d00001903sv*sd*bc*sc*i*
alias:          pci:v00001924d00000903sv*sd*bc*sc*i*
alias:          pci:v00001924d00000813sv*sd*bc*sc*i*
alias:          pci:v00001924d00000803sv*sd*bc*sc*i*
depends:        mdio,mtd
name:           sfc
vermagic:       4.18.0-305.12.1.el8_4.x86_64 SMP mod_unload modversions
parm:           major:char device major number to use (int)
parm:           efx_allow_nvconfig_writes:Allow access to static config and backup firmware (bool)
parm:           enable_vswitch:Force allocation of a VEB vswitch on supported adapters (bool)
parm:           vfs_vlan_restrict:[SFC9100-family] Restrict VLANs usage on VFs. VF driver needs to use HW VLAN filtering to get VLAN tagged traffic; default=N (bool)
parm:           vf_tx_filter:[SFC9000-family] VF transmit filter mode 0=>never, 1=>auto(default), 2=>always (int)
parm:           vf_max_tx_channels:[SFC9000-family] Limit the number of TX channels VFs can use (uint)
parm:           max_vfs:Specify the number of VFs initialized by the driver (array of int)
parm:           vf_count:Duplicate of the max_vfs parameter (array of int)
parm:           mcdi_logging_default:Enable MCDI logging on newly-probed functions (bool)
parm:           rx_recycle_ring_size:Maximum number of RX buffers to recycle pages for (uint)
parm:           underreport_skb_truesize:Give false skb truesizes. Debug option to restore previous driver behaviour. (bool)
parm:           rx_copybreak:Maximum size of packet that may be copied directly to the network stack
parm:           rx_refill_threshold:RX descriptor ring refill threshold (%) (uint)
parm:           lro_table_size:Size of the LRO hash table.  Must be a power of 2 (uint)
parm:           lro_chain_max:Maximum length of chains in the LRO hash table (uint)
parm:           lro_idle_jiffies:Time (in jiffies) after which an idle connection's LRO state is discarded (uint)
parm:           lro_slow_start_packets:Number of packets that must pass in-order before starting LRO. (uint)
parm:           lro_loss_packets:Number of packets that must pass in-order following loss before restarting LRO. (uint)
parm:           tx_copybreak:Maximum size of packet that may be copied to a new buffer on transmit, minimum is 16 bytes or 0 to disable (uint)
parm:           piobuf_size:[SFC9100-family] Maximum size of packet that may be copied to a PIO buffer on transmit (uint)
parm:           tx_push_max_fill:[SFC9100-family] Only use Tx push when the queue is below this fill level; 0=>never push 1=>push when empty; default always to push (uint)
parm:           tx_coalesce_doorbell:[SFC9100-family] Coalesce notification to NIC of pending TXdata when set this option sets tx_push_max_fill=0::default=N (bool)
parm:           multicast_chaining:[SFC9100-family] Enabled multicast filter chaining in firmware; default=Y (bool)
parm:           monitor_hw_available:[SFC9100-family] Check hardware availability during periodic monitor; default=N (bool)
parm:           tx_non_csum_queue:[SFC9100-family] Allocate dedicated TX queues for traffic not requiring checksum offload; default=N (bool)
parm:           rx_hash_insert:[SFC9000-family] Enable insertion of flow hash in RX buffers (bool)
parm:           rx_desc_cache_size:[SFC4000/SFC9000-family] Set RX descriptor cache size (int)
parm:           tx_desc_cache_size:[SFC4000/SFC9000-family] Set TX descriptor cache size (int)
parm:           efx_disable_port_on_mpe:[SFC4000/SFC9000-family] Disable interface on Memory Parity Error (bool)
parm:           lro:Large receive offload acceleration (bool)
parm:           separate_tx_channels:Use separate channels for TX and RX (bool)
parm:           xdp_alloc_tx_resources:[EXPERIMENTAL] Allocate resources for XDP TX (bool)
parm:           monitor_interval_ms:Bus state test interval in ms (uint)
parm:           rss_cpus:Number of CPUs to use for Receive-Side Scaling, or 'packages', 'cores', 'hyperthreads', 'numa_local_cores' or 'numa_local_hyperthreads' (charp)
parm:           rss_numa_local:Restrict RSS to CPUs on the local NUMA node (bool)
parm:           rss_use_fixed_key:Use a fixed RSS hash key, tested for reliable spreading across channels (bool)
parm:           irq_adapt_enable:Enable adaptive interrupt moderation (bool)
parm:           irq_adapt_low_thresh:Threshold score for reducing IRQ moderation (uint)
parm:           irq_adapt_high_thresh:Threshold score for increasing IRQ moderation (uint)
parm:           irq_adapt_irqs:Number of IRQs per IRQ moderation adaptation (uint)
parm:           debug:Bitmapped debugging message enable value (uint)
parm:           rx_ring:Maximum number of descriptors in a receive ring (uint)
parm:           tx_ring:Maximum number of descriptors in a transmit ring (uint)
parm:           num_vis:Set number of VIs (int)
parm:           phy_power_follows_link:Power down phy when interface is administratively down (bool)
parm:           performance_profile:Tune settings for different performance profiles: 'throughput', 'latency' or (default) 'auto' (charp)
parm:           auto_config_xps:Toggle automatic XPS configuration (default is enabled). (bool)
parm:           irq_set_affinity:Set SMP affinity of IRQs to support RSS (N=>disabled Y=>enabled (default)) (bool)
parm:           napi_weight:NAPI weighting (int)
parm:           rx_irq_mod_usec:Receive interrupt moderation (microseconds) (uint)
parm:           tx_irq_mod_usec:Transmit interrupt moderation (microseconds) (uint)
parm:           interrupt_mode:Interrupt mode (0=>MSIX 1=>MSI 2=>legacy) (uint)
```

and  when i type  `ip addr ` got no `sfc` drive ,  
```
[somewheve@localhost ~]$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: enp7s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether f0:2f:74:95:5d:f6 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.102/24 brd 192.168.1.255 scope global dynamic noprefixroute enp7s0
       valid_lft 5894sec preferred_lft 5894sec
    inet6 fe80::f22f:74ff:fe95:5df6/64 scope link noprefixroute
       valid_lft forever preferred_lft forever
3: enp8s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether f0:2f:74:95:5d:f5 brd ff:ff:ff:ff:ff:ff
4: virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 52:54:00:70:ff:a9 brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
       valid_lft forever preferred_lft forever
5: virbr0-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc fq_codel master virbr0 state DOWN group default qlen 1000
    link/ether 52:54:00:70:ff:a9 brd ff:ff:ff:ff:ff:ff
```

can you help with this? I got no idea about this.

**Comment by somewheve** [comment]
my network card go died... sorry for this
